rooms.js

```diff
@@ -163,6 +165,95 @@ var Room = (function () {
 		}
 		return true;
 	};
+	//mute handling
+	Room.prototype.runMuteTimer = function () {
+		if (this.muteTimer) return;
+		if (this.mutedUsers[0].time < Date.now() + 5) {
+			// this prevents lag from bugging the timer
+			this.mutedUsers[0].time = Date.now() + 5;
+		}
+		var self = this;
+		this.muteTimer = setTimeout(function () {
+			var user = Users(self.mutedUsers[0].userid);
```

Comment:
Does it really need to be it's own function? What if instead I did something like:
```
if (this.mutedUsers[0].time <= Date.now()) {
    var user = Users(self.mutedUsers[0].userid);
    if (user) {
        user.popup("Your mute has expired.");
        this.unmute(user);
    }
} else {
    ... [timer related stuff] ...
}
```